src/sentry/integrations/github/client.py

```diff
@@ -174,7 +184,7 @@ def get_with_pagination(self, path: str, response_key: str | None = None) -> Seq
 
             # XXX: In order to speed up this function we will need to parallelize this
             # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
-            while get_next_link(resp) and page_number < self.page_number_limit:
+            while get_next_link(resp) and page_number < page_number_limit:
```

Comment:
This is good. Just a bit of info about this block.

page_limit was being used just in case GH would start giving us garbage.
`get_next_link` is the logic handling pagination. Page number is just a safety measure.

Now with your change it will actually be useful!